extension/src/countminsketch.rs

```diff
@@ -0,0 +1,143 @@
+use pgx::*;
+
+use aggregate_builder::aggregate;
+use countminsketch::{CountMinHashFn, CountMinSketch as CountMinSketchInternal};
+use flat_serialize::*;
+
+use crate::{
+    flatten,
+    palloc::{Inner, Internal},
+    pg_type,
+    raw::bytea,
+    ron_inout_funcs,
+};
+
+#[pg_schema]
+pub mod toolkit_experimental {
+    use super::*;
+
+    pg_type! {
+        #[derive(Debug)]
+        struct CountMinSketch<'input> {
+            width: u32,
+            depth: u32,
+            hash_keys: [u64; self.depth],
+            counters: [i64; self.width * self.depth],
+        }
+    }
+
+    impl CountMinSketch<'_> {
+        pub fn new(width: u32, depth: u32, hash_keys: Vec<u64>, counters: Vec<i64>) -> Self {
```

Comment:
This seems like it's only useful in `from_internal_countminsketch`.  If that's the case it shouldn't be public.